hazelcast/src/main/java/com/hazelcast/map/impl/operation/MergeOperation.java

```diff
@@ -161,16 +170,34 @@ public int getAsyncBackupCount() {
     @Override
     protected void afterRunInternal() {
         invalidateNearCache(invalidationKeys);
+
+        super.afterRunInternal();
     }
 
     @Override
     public Operation getBackupOperation() {
-        return new PutAllBackupOperation(name, backupRecordAndDataValuePairs, disableWanReplicationEvent);
+        return new PutAllBackupOperation(name,
+                toBackupListByRemovingEvictedRecords(), disableWanReplicationEvent);
+    }
+
+    @Nonnull
+    private List toBackupListByRemovingEvictedRecords() {
```

Comment:
Why did we move collecting the records to this stage? Is it because records may get evicted on NOOME after they have been merged in `MergeOperation#merge`?

Reply:
right.